lib/src/path_parser.cc

```diff
@@ -143,14 +143,18 @@ namespace hocon {
             return token_list { t };
         }
 
+        // The split iterator tacks an empty string onto the end of the iterator if the string
+        // ends in the split character; so we need to remove trailing '.'
+        string text_to_split = boost::trim_right_copy_if(token_text, is_any_of("."));
+
         auto token_it = boost::make_iterator_range(make_split_iterator(
-                token_text, token_finder(is_any_of("."))), split_iterator<string::const_iterator>());
+                text_to_split, token_finder(is_any_of("."))), split_iterator<string::iterator>());
 
         token_list split_tokens;
         for (auto& s : token_it) {
             string token_string { s.begin(), s.end() };
```

Comment:
Is this really needed? I'd think s would implicitly convert to string as needed for the make_shared calls.